backend/Modules/Users/HitMeApp.Users/Infrastructure/IUserRepository.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using HitMeApp.Shared.DDD;
+using HitMeApp.Users.Models;
+
+namespace HitMeApp.Users.Infrastructure
+{
+    internal interface IUserRepository : IRepository
+    {
+        public Task Add(User user);
+        public bool Exists(string email);
```

Comment:
`bool` -> `Task<bool>`. Unfortunatelly most of the repositories will be called asynchronously, and it kind of enforces the signature to follow the pattern. It's easier to handle asynchronous interfaces synchronously than the other way around.